test/test_suite_runner.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+
+from test_suite.suite_runner import SuiteRunner
+
+
+class TestSuiteRunner:
+    test_cloud_provider = 'aws'
+    test_instances = []
+    test_ssh_identity_file = '/path/to/ssh/key.pem'
+    test_output_filepath = 'test/output/filepath.xml'
+
+    @pytest.fixture
+    def suite_runner(self):
+        return SuiteRunner(self.test_cloud_provider,
+                           self.test_instances,
+                           self.test_ssh_identity_file)
+
+    def test_run_tests(self, mocker, suite_runner):
+        # Arrange
+        test_output_filepath = '/test/output/filepath'
+        test_composed_command = 'test composed command'
+
+        mock_compose_testinfra_command = mocker.MagicMock(return_value=test_composed_command)
+        suite_runner.compose_testinfra_command = mock_compose_testinfra_command
+
+        mock_os_system = mocker.patch('os.system')
+
+        # Act
+        suite_runner.run_tests(test_output_filepath)
+
+        # Assert
+        mock_compose_testinfra_command.assert_called_once_with(test_output_filepath)
+        mock_os_system.assert_called_once_with(test_composed_command)
+
+    @pytest.mark.parametrize(
+        'test_debug, test_parallel, expected_command_string',
+        [(False, False,
+          f'py.test path1 path2 --hosts=user1@host1,user2@host2 '
+          f'--ssh-identity-file {test_ssh_identity_file} --junit-xml {test_output_filepath}'),
+         (False, True,
+          f'py.test path1 path2 --hosts=user1@host1,user2@host2 '
+          f'--ssh-identity-file {test_ssh_identity_file} --junit-xml {test_output_filepath} '
+          f'--numprocesses=logical --dist=loadfile'),
+         (True, True,
+          f'py.test path1 path2 --hosts=user1@host1,user2@host2 '
+          f'--ssh-identity-file {test_ssh_identity_file} --junit-xml {test_output_filepath} '
+          f'-v '
+          f'--numprocesses=logical --dist=loadfile')]
+    )
+    def test_compose_testinfra_command(self,
+                                       mocker,
+                                       suite_runner,
+                                       test_debug,
+                                       test_parallel,
+                                       expected_command_string):
+        # Arrange
+        test_hosts = 'user1@host1,user2@host2'
+        test_suite_paths = ['path1', 'path2']
+
+        suite_runner.debug = test_debug
+        suite_runner.parallel = test_parallel
+
+        mock_get_all_instances_hosts_with_users = mocker.MagicMock(return_value=test_hosts)
+        suite_runner.get_all_instances_hosts_with_users = mock_get_all_instances_hosts_with_users
+
+        mock_get_test_suite_paths = mocker.MagicMock(return_value=test_suite_paths)
+        suite_runner.get_test_suite_paths = mock_get_test_suite_paths
+
+        # Act, Assert
+        assert suite_runner.compose_testinfra_command(self.test_output_filepath) == expected_command_string
```

Comment:
Actually if we mix up the order we can end up with a command execution error. It's true that it may affect only to a couple of command options and not all of them. However, I think unit tests should be as specific as possible to test even if the array contained the correct order. Specially if these items are added "manually" in that order in the code.